stix2/sources/__init__.py

```diff
@@ -557,150 +687,198 @@ def add_filter(self, filters):
-                    #         clean = False
-                    #         break
-                    #
-                    # elif filter_['op'] == "<=":
-                    #     if not stix_obj[filter_['field']] <= filter_['value']:
-                    #         clean = False
-                    #         break
-
-                except TypeError:
-                    # type mismatch of comparison operands - ignore filter,
-                    # no error raised for now
-                    pass
-
-            # if object unmarked after all filter, add it
-            if clean:
-                filtered_stix_objs.append(stix_obj)
-
-            clean = True
-
-        return filtered_stix_objs
+        return copy.deepcopy(list(self.filters.values()))
 
     def deduplicate(self, stix_obj_list):
-        """deduplicate a list of STIX objects into a unique set
+        """deduplicate a list fo STIX objects to a unique set
 
-        reduces a set of STIX objects to unique set by looking
+        Reduces a set of STIX objects to unique set by looking
         at 'id' and 'modified' fields - as a unique object version
         is determined by the combination of those fields
 
         Args:
             stix_obj_list (list): list of STIX objects (dicts)
 
         Returns:
-            (list): a unique set of the passed STIX object list
-
-
+                (list): unique set of the passed list of STIX objects
         """
+
         unique = []
-        have = False
+        dont_have = False
         for i in stix_obj_list:
+            dont_have = False
             for j in unique:
-                if i['id'] == j['id'] and i['modified'] == j['modified']:
-                    have = True
+                for field in STIX_VERSION_FIELDS:
+                    if not i[field] == j[field]:
+                        dont_have = True
                     break
-            if not have:
+            if dont_have:
                 unique.append(i)
-            have = False
         return unique
+
+
+class STIXCommonPropertyFilters():
+    """
+    """
+    @classmethod
+    def _all(cls, filter_, stix_obj_field):
+        """all filter operations (for filters whose value type can be applied to any operation type)"""
+        if filter_["op"] == '=':
+            return stix_obj_field == filter_["value"]
+        elif filter_["op"] == "!=":
+            return stix_obj_field != filter_["value"]
+        elif filter_["op"] == "in":
+            return stix_obj_field in filter_["value"]
+        elif filter_["op"] == ">":
+            return stix_obj_field > filter_["value"]
+        elif filter_["op"] == "<":
+            return stix_obj_field < filter_["value"]
+        elif filter_["op"] == ">=":
+            return stix_obj_field >= filter_["value"]
+        elif filter_["op"] == "<=":
+            return stix_obj_field <= filter_["value"]
+        else:
+            return -1
+
```

Comment:
Yeah, [functions should be organized into modules instead of classes](https://github.com/amontalenti/elements-of-python-style#functions-vs-classes). I like the dict approach, too - would make it easier for a user to define a filter type for their custom properties if necessary.